cmd/cluster/openstack/create.go

```diff
@@ -0,0 +1,245 @@
+				Name: o.OpenStackExternalNetworkName,
+			},
+		}
+	}
+
+	cluster.Spec.Services = core.GetIngressServicePublishingStrategyMapping(cluster.Spec.Networking.NetworkType, o.externalDNSDomain != "")
+	if o.externalDNSDomain != "" {
+		for i, svc := range cluster.Spec.Services {
+			switch svc.Service {
+			case hyperv1.APIServer:
+				cluster.Spec.Services[i].Route = &hyperv1.RoutePublishingStrategy{
+					Hostname: fmt.Sprintf("api-%s.%s", cluster.Name, o.externalDNSDomain),
+				}
+
+			case hyperv1.OAuthServer:
+				cluster.Spec.Services[i].Route = &hyperv1.RoutePublishingStrategy{
+					Hostname: fmt.Sprintf("oauth-%s.%s", cluster.Name, o.externalDNSDomain),
+				}
+
+			case hyperv1.Konnectivity:
+				cluster.Spec.Services[i].Route = &hyperv1.RoutePublishingStrategy{
+					Hostname: fmt.Sprintf("konnectivity-%s.%s", cluster.Name, o.externalDNSDomain),
+				}
+			case hyperv1.Ignition:
+				cluster.Spec.Services[i].Route = &hyperv1.RoutePublishingStrategy{
+					Hostname: fmt.Sprintf("ignition-%s.%s", cluster.Name, o.externalDNSDomain),
+				}
+			case hyperv1.OVNSbDb:
+				cluster.Spec.Services[i].Route = &hyperv1.RoutePublishingStrategy{
+					Hostname: fmt.Sprintf("ovn-sbdb-%s.%s", cluster.Name, o.externalDNSDomain),
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (o *RawCreateOptions) GenerateNodePools(constructor core.DefaultNodePoolConstructor) []*hyperv1.NodePool {
+	return nil
+}
+
+func (o *CreateOptions) GenerateResources() ([]client.Object, error) {
+	credentialsContents, err := os.ReadFile(o.OpenStackCredentialsFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read OpenStack credentials file: %w", err)
+	}
+	credentialsSecret := credentialsSecret(o.namespace, o.name)
+	credentialsSecret.Data = map[string][]byte{
+		"clouds.yaml": credentialsContents,
+	}
+
+	if o.OpenStackCACertFile != "" {
+		caCertContents, err := os.ReadFile(o.OpenStackCACertFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read OpenStack CA certificate file: %w", err)
+		}
+		credentialsSecret.Data["cacert"] = caCertContents
+	}
+
+	return []client.Object{credentialsSecret}, nil
+}
+
+func credentialsSecret(namespace, name string) *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name + "-cloud-credentials",
+			Namespace: namespace,
+			Labels:    map[string]string{util.DeleteWithClusterLabelName: "true"},
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+}
+
+var _ core.Platform = (*CreateOptions)(nil)
+
+func NewCreateCommand(opts *core.RawCreateOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "openstack",
+		Short:        "Creates basic functional HostedCluster resources on OpenStack platform",
+		SilenceUsage: true,
+	}
+
+	openstackOpts := DefaultOptions()
+	BindOptions(openstackOpts, cmd.Flags())
+	cmd.MarkPersistentFlagRequired("pull-secret")
```

Comment:
You may need to validate this flag is not empty in your validation function. We recently refactored how the CLI works and I just noticed this doesn't seem to be marking the flags as required - [OCPBUGS-37323](https://issues.redhat.com/browse/OCPBUGS-37323).

Reply:
nice suggestion, done. Thanks